app.ts

```diff
@@ -0,0 +1,243 @@
+					window.innerHeight <= 646;
+					break;
+				case 1:
+					window.innerHeight <= 969;
+					break;
+				case 2:
+					window.innerHeight <= 1938;
+					break;
+				default:
+					recordDisplayset = 15;
+					break;
+			}
+			return recordDisplayset + possibleId;
+		}
+		function nextPageResize(previous: number[]): number {
+			const fromID = ConvertNumber(previous.sort((a, b) => { return a - b })[0]);
+			const toID = ConvertNumber(previous.sort((a, b) => { return a - b })[1]);
+			const documentHeight = $(window).innerHeight() as number - ($(`#table-row-${fromID}`).height() as number);
+			for (let i = fromID; i <= toID; i++) {
+				const elementHeightOffset = ($(`#table-row-${i}`).offset() as JQueryCoordinates).top;
+				if (elementHeightOffset < documentHeight) continue;
+				return i;
+			}
+			return toID;
+		}
+		function previousPageResize(previous: number[]): number[] {
+			const toId = calculateToId(previous[0] - (nextPageResize(previous) - previous[0]));
+			return [previous[0] - (nextPageResize(previous) - previous[0]), toId];
+		}
+		window.onload = async () => {
+
+			// trigger async function
+			// log response or catch error of fetch promise
+			getColumnNamesCall().then(data => console.log(data)).catch(reason => console.log(reason.message));
+
+			// log response or catch error of fetch promise
+			getRecordCountCall().then(data => console.log(data)).catch(reason => console.log(reason.message));
+
+			previous = await LoadPageContent(0, calculateToId(0));
+			$("#previous").click(async () => {
+				const CountData = await getRecordCountCall();
+				previous = previousPageResize(previous);
+				let fromId = previous[0] >= 0 ? previous[0] : 0;
+				const possibleStep = calculateToId(fromId) - fromId;
+				let toId = (previous[0] >= 0 ? previous[1] : possibleStep);
+				fromId = fromId == CountData - 1 ? fromId - possibleStep : fromId;
+				toId = toId <= CountData - 1 ? toId : CountData - 1;
+				previous = await LoadRecordsData(fromId, toId);
+			});
+
+			$("#next").click(async () => {
+				try {
+					const recordCount = await getRecordCountCall();
+					const fromId = nextPageResize(previous);
+					const possibleStep = calculateToId(fromId) - fromId;
+					if (fromId <= recordCount - possibleStep - 1) {
+						const toId = fromId + possibleStep <= recordCount - 1 ? fromId + possibleStep : recordCount - 1;
+						previous = await LoadRecordsData(fromId, toId);
+					} else if (fromId <= recordCount - 1) {
+
+						previous = await LoadRecordsData(recordCount - 1 - (calculateToId(fromId) - fromId), recordCount - 1);
+					} else {
+						console.log("Test is working ");
+					}
+				} catch (error) {
+				}
+			});
+			$("#go-button").click(async () => {
+				const recordCount = await getRecordCountCall();
+				const fromId = ConvertNumber($("#index").val() as string, false);
+				const possibleStep = calculateToId(fromId) - fromId;
+				if (fromId < 0) {
+					alert('only insert Id greater than or equal to 0');
```

Comment:
add a return here so we can remove the next "else" below.  This is more readable and reduces excessive indentation in cases where larger nested "if else" statements exist.

Reply:
The comment on this line is still relevant.